generic/tests/netperf.py

```diff
@@ -639,58 +644,70 @@ def parse_demo_result(fname, sessions):
+        ssh_cmd(clients[-1], "rm -f %s" % fname)
+        numa_enable = params.get("netperf_with_numa", "yes") == "yes"
+        timeout_netperf_start = int(l) * 0.5
+        client_thread = threading.Thread(
+                target=netperf_thread,
+                kwargs={"i": int(sessions),
+                        "numa_enable": numa_enable,
+                        "client_s": clients[0],
+                        "timeout": timeout_netperf_start})
+
+        client_thread.start()
+
+        ret = {}
+        ret['pid'] = pid
+
+        if utils_misc.wait_for(
+                all_clients_up,
+                timeout_netperf_start,
+                0.0, 0.2,
+                "Wait until all netperf clients start to work"):
+            logging.debug("All netperf clients start to work.")
+
+            # real & effective test starts
+            if get_status_flag:
+                start_state = get_state()
+            ret['mpstat'] = ssh_cmd(host, "mpstat 1 %d |tail -n 1" % (l - 1))
+            finished_result = ssh_cmd(clients[-1], "cat %s" % fname)
+
+            # stop netperf clients
+            kill_cmd = "killall netperf"
+            if params.get("os_type_client") == "windows":
+                kill_cmd = "taskkill /F /IM netperf*"
+            ssh_cmd(clients[-1], kill_cmd, ignore_status=True)
+
+            # real & effective test ends
+            if get_status_flag:
+                end_state = get_state()
+                if len(start_state) != len(end_state):
+                    msg = "Initial state not match end state:\n"
+                    msg += "  start state: %s\n" % start_state
+                    msg += "  end state: %s\n" % end_state
+                    logging.warn(msg)
+                else:
+                    for i in range(len(end_state) / 2):
+                        ret[end_state[i * 2]] = (end_state[i * 2 + 1] -
+                                                 start_state[i * 2 + 1])
+
+            client_thread.join()
+
+            error_context.context("Testing Results Treatment and Report",
+                                  logging.info)
+            f = open(fname, "w")
+            f.write(finished_result)
+            f.close()
+            ret['thu'] = parse_demo_result(fname, int(sessions))
+            return ret
+            break
         else:
-            for i in range(len(end_state) / 2):
-                ret[end_state[i * 2]] = (end_state[i * 2 + 1] -
-                                         start_state[i * 2 + 1])
-
-    client_thread.join()
-
-    error_context.context("Testing Results Treatment and Report", logging.info)
-    f = open(fname, "w")
-    f.write(finished_result)
-    f.close()
-    ret['thu'] = parse_demo_result(fname, int(sessions))
-    return ret
+            kill_cmd = "killall netperf"
+            ssh_cmd(clients[-1], kill_cmd, ignore_status=True)
+            tries = tries - 1
+            logging.debug("left %s times" % tries)
```

Comment:
Do we need fail the test case if all tries failed?

Reply:
It's not good to fail(quit) tests, if fail tests, other sub-tests have impossible running in job.